pkg/apply/merge.go

```diff
@@ -133,8 +128,9 @@ func mergeAnnotations(current, updated *uns.Unstructured) {
 	for k, v := range updatedAnnotations {
 		curAnnotations[k] = v
 	}
-
-	updated.SetAnnotations(curAnnotations)
+	if len(curAnnotations) > 1 {
```

Comment:
Will this block if from setting annotation to empty?

Reply:
As abovementioned, we shall keep the obj only contains the fields that we really need.